include/pstl_offload/internal/usm_memory_replacement.h

```diff
@@ -53,34 +54,59 @@ __get_offload_device()
 class __offload_policy_holder_type
 {
     using __set_active_device_func_type = void (*)(sycl::device*);
+    using __get_offload_device_func_type = sycl::device (*)();
 
   public:
     // Since the global object of __offload_policy_holder_type is static but the constructor
     // of the class is inline, we need to avoid calling static functions inside of the constructor
     // and pass the pointer to exact function as an argument to guarantee that the correct __active_device
     // would be stored in each translation unit
-    __offload_policy_holder_type(sycl::device&& __offload_device,
+    template <typename _DeviceSelector>
+    __offload_policy_holder_type(const _DeviceSelector& __device_selector,
                                  __set_active_device_func_type __set_active_device_func)
-        : _M_offload_device(std::move(__offload_device)), _M_offload_policy(_M_offload_device),
-          _M_set_active_device(__set_active_device_func)
+        : _M_set_active_device(__set_active_device_func)
     {
-        _M_set_active_device(&_M_offload_device);
+        try
+        {
+            _M_offload_device.emplace(__device_selector);
+        }
+        catch (const sycl::exception& e)
+        {
+            // e.code() == sycl::errc::runtime occurs when device selection unable to get offload device
+            // with required type. Do not pass an exception, as ctor is called for a static object and
+            // the exception can't be processed.
+            // Remember the situation and re-throw exception when asked for the policy from user's code.
+            if (e.code() != sycl::errc::runtime)
+                throw;
+        }
+
+        if (_M_offload_device.has_value())
+        {
+            _M_set_active_device(&*_M_offload_device);
```

Comment:
```suggestion
            _M_set_active_device(&_M_offload_device.value());
```
How about that? To me it's just simpler to read comparing with `&*`

Reply:
Done.